src/Container.ts

```diff
@@ -89,8 +92,16 @@ export class Container {
             params.unshift(null);
         }
 
-        // create a new instance of the requested object
-        const objectInstance = new (type.bind.apply(type, params))();
+        let objectInstance: any;
+
+        if (factory) {
+            // Calling factory function if specified to create an instance.
+            objectInstance = factory.apply(undefined, params);
```

Comment:
@pleerock Why do **we** need to pass something to the factory function? The end-user should decide what should be passed to it.